### ISSUE TITLE
Documentation about disabling master password on Mac gives wrong path, instructions

### ISSUE THREAD

**Issue by gisborne** [pgadmin-org/pgadmin4#5369]
The documentation [here](https://www.pgadmin.org/docs/pgadmin4/6.14/master_password.html) says that to disable the master password, a file at:

/Library/Preferences/pgadmin/config_system.py

needs to be changed. The initial page is confusing and can be read as suggesting to change instead maybe config.py at that location. In any event, the file should be modified to include:

MASTER_PASSWORD_REQUIRED=False

There is more than one issue:

1. The path should be ~/Library/Preferences/pgadmin/config_system.py (note the initial ~)
2. The initial file should clarify that the user is not modifying config.py nor probably config_system.py, but config_local.py. The instructions do say to read on another page about config.py; I just thing it could be clearer
3. The only file on my system at ~/Library/Preferences/pgadmin/config_system.py is runtime_config.json. I assume the user is supposed to create if need be a config_local.py at that location.
4. A suitably-corrected version of the advice given, where I put the line `MASTER_PASSWORD_REQUIRED=False` in config_local.py (I've also tried the other config*.py file names) does not produce the desired result. Quitting and re-loading PGAdmin still results in the password prompt.

**Comment by dpage** [comment]
> The documentation [here](https://www.pgadmin.org/docs/pgadmin4/6.14/master_password.html) says that to disable the master password, a file at:
> 
> /Library/Preferences/pgadmin/config_system.py
> 
> needs to be changed. The initial page is confusing and can be read as suggesting to change instead maybe config.py at that location. In any event, the file should be modified to include:

Yeah, maybe the best option here is to think of a different title for the target page, as that's where the link text comes from.

> MASTER_PASSWORD_REQUIRED=False

Yes.

> There is more than one issue:
> 
> 1. The path should be ~/Library/Preferences/pgadmin/config_system.py (note the initial ~)

That is *exactly* wrong. The whole point of config_system.py is to allow a system administrator to enforce a system-wide configuration, not one from a users home directory.

> 2. The initial file should clarify that the user is not modifying config.py nor probably config_system.py, but config_local.py. The instructions do say to read on another page about config.py; I just thing it could be clearer

The problem is that the docs discuss configuration options in (currently) 14 different places. It's not practical to have the full information repeated in so many places.

> 3. The only file on my system at ~/Library/Preferences/pgadmin/config_system.py is runtime_config.json. I assume the user is supposed to create if need be a config_local.py at that location.

Yes - the docs do say that:

![Screenshot 2022-10-03 at 10 20 36](https://user-images.githubusercontent.com/316438/193543269-2a415389-091a-4503-b950-1c8ae46d0b5e.png)

> 4. A suitably-corrected version of the advice given, where I put the line `MASTER_PASSWORD_REQUIRED=False` in config_local.py (I've also tried the other config*.py file names) does not produce the desired result. Quitting and re-loading PGAdmin still results in the password prompt.

Where did you create config_local.py?

**Comment by gisborne** [comment]
I have been running the latest PGAdmin, so it should have created its files wherever.

There is no /Library/Preferences/pgadmin/. Should there be?

Creating this directory, and within it a config_system.py, and putting the single line in it `MASTER_PASSWORD_REQUIRED=False` does disable the password prompt.

It seems odd that this should be at the system level. Might different users not want different behavior here?

Changing the file name to config_local.py restores the password prompt (ie the file is ignored). Note that I already have  ~/Library/Preferences/pgadmin/config_local.py containing `MASTER_PASSWORD_REQUIRED=False`. Is it just that this particular setting is ignored in the _local file? Does such a file have no effect generally?

**Comment by dpage** [comment]
> I have been running the latest PGAdmin, so it should have created its files wherever.
> 
> There is no /Library/Preferences/pgadmin/. Should there be?

No. You only need to create that directory and it's contents if you need to.

> 
> Creating this directory, and within it a config_system.py, and putting the single line in it `MASTER_PASSWORD_REQUIRED=False` does disable the password prompt.
> 
> It seems odd that this should be at the system level. Might different users not want different behavior here?

The whole point is that a system administrator can create a configuration for all users by creating a config_system.py. Users without root/administrator privileges cannot create/modify that directory and file. Other measures are required to make that secure of course - most importantly, ensuring the user cannot just use a private copy of pgAdmin which is modified to ignore any system-wide config, however it is also useful for deploying a managed configuration for convenience as well.

> Changing the file name to config_local.py restores the password prompt (ie the file is ignored). Note that I already have ~/Library/Preferences/pgadmin/config_local.py containing `MASTER_PASSWORD_REQUIRED=False`. Is it just that this particular setting is ignored in the _local file? Does such a file have no effect generally?

Right. Only config_system.py will work in the global location. And config_local.py & config_distro.py are expected to be in the same directory as the default config.py. Only the runtime config file is expected to be in ~/Library/Preferences/pgadmin/, and that shouldn't ever need to be manually editted unless you're a pgAdmin developer.

**Comment by gisborne** [comment]
I would expect that having no /Library/Preferences/pgadmin/config_system.py and a /Library/Preferences/pgadmin/config_local.py would let me set this just for myself, since there is no global setting, but it doesn't work?

**Comment by gplanansky** [comment]
On macOS 13.1, with pgAdmin 4  version 6.19 , putting the `config_local.py` file in the `Applications/pgAdmin\ 4.app/Contents/Resources/web/` directory works.  Thus:

   `$cat /Applications/pgAdmin\ 4.app/Contents/Resources/web/config_local.py`
   `MASTER_PASSWORD_REQUIRED = False`

However that location is not properly documented. Worse, updating the app blows away any files put therein.  

It would be good to have the user's local config file in the user's home directory, say `~/.pgadmin4/config.py` or `~/.config/pgadmin4/config.py`.  Or make those `config_local.py` to preserve the "local" appellation.    

That would follow common practice and keep users from diddling with the app hierarchy.  Importantly, it would keep updates from blowing away user files.

Regarding administrative practices,  pgadmin is often used in instructional settings, e.g. short courses, perhaps on the road, where passwords can trip up a class at the outset, keep it from having a clean start.  Having config files that go in user's home directories can solve many such problems.  It's a significant use case to consider supporting,

**Comment by khushboovashi** [comment]
You should use config_system.py file for the admin level config settings. It will not be updated on app update as it will be outside of the pgAdmin installation.
Please refer https://www.pgadmin.org/docs/pgadmin4/6.19/config_py.html for more details.

**Comment by gisborne** [comment]
My point is that the documentation should be clearer. At least it shouldn’t be wrong.
